sys/net/gnrc/network_layer/ipv6/nib/nib.c

```diff
@@ -1733,4 +1797,46 @@ static uint32_t _handle_rio(gnrc_netif_t *netif, const ipv6_hdr_t *ipv6,
 
     return route_ltime;
 }
+
+#if IS_ACTIVE(CONFIG_GNRC_IPV6_NIB_QUEUE_PKT)
+gnrc_pktqueue_t *_nbr_pop_pkt(_nib_onl_entry_t *node)
+{
+    if (node->pktqueue_len == 0) {
+        assert(node->pktqueue == NULL);
+        return NULL;
+    }
+    assert(node->pktqueue != NULL);
+
+    node->pktqueue_len--;
+
+    return gnrc_pktqueue_remove_head(&node->pktqueue);
+}
+
+void _nbr_push_pkt(_nib_onl_entry_t *node, gnrc_pktqueue_t *pkt)
+{
+    static_assert(CONFIG_GNRC_IPV6_NIB_NBR_QUEUE_CAP <= UINT8_MAX,
+                  "nib: nbr queue cap overflows counter");
+    assert(_get_nud_state(node) == GNRC_IPV6_NIB_NC_INFO_NUD_STATE_INCOMPLETE);
+
+    if (ARRAY_SIZE(_queue_pool) > CONFIG_GNRC_IPV6_NIB_NBR_QUEUE_CAP &&
+        node->pktqueue_len == CONFIG_GNRC_IPV6_NIB_NBR_QUEUE_CAP) {
```

Comment:
```suggestion
    if (node->pktqueue_len == (ARRAY_SIZE(_queue_pool) - 1) ||
        node->pktqueue_len == CONFIG_GNRC_IPV6_NIB_NBR_QUEUE_CAP) {
```

I thought you wanted to prevent one node from hogging the entire queue?

Reply:
we're talking past each other here. Let's recap:
 - we have `CONFIG_GNRC_IPV6_NIB_NUMOF + 1` queue entries in the cache s.t. when they're used up there's always one neighbor with at least two
 - with `CONFIG_GNRC_IPV6_NIB_NUMOF` small ( ~ 16) we really don't care how they are distributed. Once we're out of entries, find the hog and take from there. This is fast, because we have at most ~16 neighbors and at most ~16 entries. The check `ARRAY_SIZE(_queue_pool) > CONFIG_GNRC_IPV6_NIB_NBR_QUEUE_CAP` compiles the cap-check away.
 - with  `CONFIG_GNRC_IPV6_NIB_NUMOF` big, we want a per-neighbor cap, which defaults at 16, out of the reasons stated previously: 
    - A single hog cannot deplete the entry cache by itself -> less cases where we have to iterate all neighbors to find a hog
    - RFC recommends keeping the per-neighbor queue small (whatever is considered small)

Reply:
Ah makes perfect sense!
But since I had trouble getting it the first time, better also add a comment to the code to prevent misunderstandings with the next one reading this.